optuna/_hypervolume/hssp.py

```diff
@@ -94,19 +92,18 @@ def _solve_hssp_on_unique_loss_vals(
 
     # The following logic can be used for non-unique rank_i_loss_vals as well.
     diff_of_loss_vals_and_ref_point = reference_point - rank_i_loss_vals
-    assert subset_size <= rank_i_indices.size
-    n_objectives = reference_point.size
+    (n_solutions, n_objectives) = rank_i_loss_vals.shape
+    assert isinstance(n_solutions, int), "MyPy Redefinition for NumPy v2.2.0."
     contribs = np.prod(diff_of_loss_vals_and_ref_point, axis=-1)
     selected_indices = np.zeros(subset_size, dtype=int)
     selected_vecs = np.empty((subset_size, n_objectives))
-    indices: np.ndarray[tuple[int, ...], np.dtype[Any]] = np.arange(
```

Comment:
This typing is also not correct, because `indices` is always a 1d array.
As this typing says `indices` can be a ND array, which is not correct, I removed this.